### DISCUSSION TITLE
Neovim Copilot: Could not determine Node.js version

### DISCUSSION THREAD

**Discussion by ghost** [community/community#19009] [Copilot Conversations]
Hi there!

I am setting up Copilot on my Neovim editor, but having an issue.
I followed the instructions on https://github.com/github/copilot.vim; however, when I ran `:Copilot setup` in NeoVim, I got the following error:

`Copilot: 'Could not determine Node.js version'`

Running `:let g:copilot_ignore_node_version = 1` before `:Copilot setup` skips the version checking (implemented in `autoload/copilot/agent.vim`), but then again Neovim just hangs executing `:Copilot setup`.

Here is my environment:
* Ubuntu 16.04
* Neovim v0.7.0
* Node.js 16.15.1 (installed via `snap`, `node --version` runs w/o error in the same session as `nvim`)

Anyone else having the error? Any help would be appreciated!

**Comment by tpope** [comment]
Could you check `:echo system(['node', '--version'])`? That's a bit closer to (but not exactly like) how Copilot itself runs it.

**Comment by ghost** [comment]
Thank you for your reply! Here is the result (with additional experiments):
```
(in Neovim)
:echo system(['node', '--version'])
-> emits nothing (as opposed to when I run node --version on bash)

:echo system(['nvim', '--version'])
-> does not yield anything, either

:echo system(['vim', '--version'])
-> returns `VIM - Vi IMproved 7.4` etc.
```

Now I can see the issue: Neovim cannot capture the output of some commands. It's weird that, in Vim, `!node --version` and `!nvim --version` work as expected...
While this issue does not seem to be Copilot-specific, I will update this thread if I find a fix.

**Comment by veryhappythings** [comment]
This is not exactly a solution, but I found that this occurred when I used `snap` to install node, but replacing it with [nodenv](https://github.com/nodenv/nodenv) fixed my problem. I found that heroku installed via snap had the same issue, but neovim installed via snap did not - neovim can see `nvim --version` but not `heroku --version` or `node --version`.

**Reply by QuantumFillet** [reply]
Yeah, I guess it's this bug https://github.com/orgs/community/discussions/16298#discussioncomment-2848831

**Reply by veryhappythings** [reply]
Hmm, I don't think so - I used the same version, just installed through nodenv instead of snap 🤷

**Comment by AlexBeauchemin** [comment]
For anyone stumbling upon this, I had the same issue which was caused by an invalid `package.json` file in my project (a missing coma somewhere made the whole JSON invalid)
And because I'm using [Volta](https://volta.sh/) as a node version manager, it wasn't able to detect the node version from package.json and failed to load nodejs correctly. 

Fixing package.json resolved the issue

**Comment by Karl-EdwardJeanMehu** [comment]
I ran into this issue recently. In my case, it was not due to a package.json file. Upgrading from nvim v0.10 to nvim v0.11.4 fixed this for me.

**Reply by Raaghav-Arya** [reply]
Am on nvim v0.11.4, still facing this issue.
Node installed through snap (v22.20.0).

Am getting the following error:

[Copilot.lua] Could not determine Node.js version

**Reply by hxYuki** [reply]
I will get this error when neovim is launched with a new terminal window by some other apps. Neovim itself cannot read the `node`.

I doubt this is something about environment variable.

**Comment by dominik-rls** [comment]
I use nvm and lazily resolve the version, so it breaks when node is a zsh function rather than a symlinked executable until invokation. I guess that is on me.
